baseplate/observers/metrics.py

```diff
@@ -44,15 +45,18 @@ def __init__(self, batch: metrics.Batch, server_span: Span):
         self.batch = batch
         self.base_name = "server." + server_span.name
         self.timer = batch.timer(self.base_name)
+        self.sample_timer = batch.timer_sampling_rate == 1.0 or random() < batch.timer_sampling_rate
```

Comment:
I'm still a bit concerned about doing the randomness multiple times during the same request. This could lead to some weird counting vs. if we sample at the whole request level.

Since we can scale the counters with the `sample_rate` parameter, what do you think of the proposal to do this sampling in the root `BaseplateObserver` and not even registering the sub-observers when sampling's off? Then we get consistent whole-request level sampling and do less work in the case of unsampled requests.

Reply:
Agreed on this. Let's sample the whole request.